imap/imapparse.c

```diff
@@ -575,6 +575,131 @@ static int get_search_date(struct protstream *pin, time_t *start, time_t *end)
     return EOF;
 }
 
+/*
+ * Parse a list of mailboxes
+ */
+EXPORTED int get_search_source_mboxes(struct protstream *pin,
+                                      struct protstream *pout,
+                                      struct searchargs *searchargs,
+                                      strarray_t *mboxes)
+{
+    int c, multi = 1;
+    char *intname;
+    struct buf extname = BUF_INITIALIZER;
+
+    c = prot_getc(pin);
+    if (c != '(') {
+        prot_ungetc(c, pin);
+        multi = 0;
+    }
+
+    do {
+        c = getastring(pin, pout, &extname);
+        if (!buf_len(&extname)) goto bad;
+
+        intname = mboxname_from_external(buf_cstring(&extname),
+                                         searchargs->namespace,
+                                         searchargs->userid);
+        strarray_appendm(mboxes, intname);
```

Comment:
this could crash later if intname is NULL (strarray_appendm does not check for NULL string)

Reply:
Fixed.